clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -655,15 +656,15 @@ private void adjustSequencesDueToFailedBatch(ProducerBatch batch) {
     }
 
     private void startSequencesAtBeginning(TopicPartition topicPartition) {
-        final AtomicInteger sequence = new AtomicInteger(0);
```

Comment:
I know this one can definitely be replaced since I just reviewed it, but after a very brief search this seems to be the only obvious place also..